src/main/java/com/theelixrlabs/healthcare/controller/DoctorController.java

```diff
@@ -59,4 +64,21 @@ public ResponseEntity<?> getDoctorById(@PathVariable(DoctorConstants.PATH_VARIAB
         DoctorDto doctorDto = doctorService.getDoctorById(doctorId);
         return new ResponseEntity<>(new SuccessResponse<>(true, doctorDto), HttpStatus.OK);
     }
+
+    /**
+     * Retrieves a list of doctors by their name.
+     *
+     * @param doctorName the name of the doctor to search for
+     * @return a ResponseEntity containing a SuccessResponse with a list of matching DoctorDto objects
+     */
+    @GetMapping(DoctorConstants.GET_DOCTORS_BY_NAME_ENDPOINT)
+    public ResponseEntity<SuccessResponse<List<DoctorDto>>> getDoctorsByName
+    (@RequestParam(DoctorConstants.DOCTOR_NAME_PARAM) String doctorName) {
+        doctorModelValidator.validateDoctorName(doctorName);
+        List<DoctorDto> doctorDtoList = doctorService.searchDoctorByName(doctorName);
+        if (doctorDtoList.isEmpty()) {
+            throw new CustomException(MessageConstants.NO_DOCTOR_FOUND, messageSource);
+        }
```

Comment:
It is Bad Request, if Doctor is **not** present in database?

Reply:
yes, changed into Resource not found (404).